wordpress-mobile/WordPress-iOS
# WordPressKit: Swift 4.0 Support

### Details:
This PR is a follow up to #8159 (and should be reviewed after the former is merged). We'll introduce a few changes that make the WordPressKit Target Swift 4.0 friendly.

Needs Review: @diegoreymendez 
Thanks in advance!

### To test:
1. Verify that the app builds as usual
2. Toggle WordPressShared an WordPressKit to Swift 4.0
3. Verify that the app builds!
4. Verify that the WordPressKit Unit Tests are green
